glotaran/parameter/parameter.py

```diff
@@ -144,6 +148,55 @@ def from_list_or_value(
             param._set_options_from_dict(options)
         return param
 
+    @classmethod
+    def from_dict(cls, parameter_dict: dict[str, Parameter]) -> Parameter:
+        """Create a :class:`Parameter` from a dictionary.
+
+        Expects a dictionary created by :method:`Parameter.as_dict`.
+
+        Parameters
+        ----------
+        parameter_dict : dict
+            The source dictionary.
+
+        Returns
+        -------
+        Parameter
+            The created :class:`Parameter`
+        """
+        parameter_dict = {k.replace("-", "_"): v for k, v in parameter_dict.items()}
+        parameter_dict["full_label"] = parameter_dict["label"]
+        parameter_dict["label"] = parameter_dict["label"].split(".")[-1]
+        return cls(**parameter_dict)
+
+    def as_dict(self, as_optimized: bool = True) -> dict:
```

Comment:
Wrong typing. Also check consistency by doing as_dict(from_dict(x))

Reply:
"Fixed" in 053e356